### DISCUSSION TITLE
Nextpush disconnecting issues with NC AIO

### DISCUSSION THREAD

**Discussion by gockar** [nextcloud/all-in-one#4295] [Ideas] [answered]
Also discussed in https://codeberg.org/NextPush/nextpush-android/issues/4:
To setup NextPush in NC AIO, installing UnifiedPush Provider app (https://apps.nextcloud.com/apps/uppush) and connecting the NextPush Android app works out of the box, but after some minutes it's disconnecting. 

It's probably because I can't simply increase the webserver timeout in the AIO setup, as given in the guide (https://codeberg.org/NextPush/uppush#installation)?! 

Apache: - VirtalHost Block
```
ProxyTimeout 600
```
Apache (php configuration): - VirtalHost Block
```
<Proxy "fcgi://localhost/" disablereuse=on flushpackets=on max=10>
</Proxy>
```

How can this be done properly in a basic (without additional reverse proxy) AIO (docker compose) setup? Does anyone have experience of it working or a reliable alternative config? Thanks!

**Comment by szaimen** [comment, marked as answer]
Hi, the proxytimeout in the apache conf is by default set to 3600 so should ne enough: https://github.com/nextcloud/all-in-one/blob/bf307eb5ff4701a21ab2cb4ba5fb7f31171b62f9/Containers/apache/nextcloud.conf#L51

---

> Apache (php configuration): - VirtalHost Block
> 
> ```
> <Proxy "fcgi://localhost/" disablereuse=on flushpackets=on max=10>
> </Proxy>
> ```

As for this, we would need something comparable for SetHandler: https://github.com/nextcloud/all-in-one/blob/bf307eb5ff4701a21ab2cb4ba5fb7f31171b62f9/Containers/apache/nextcloud.conf#L14
So if uppush could provide something appropriate we would be willing to do something about it.

**Reply by gockar** [reply]
Thank you, I placed your request, because I can't answer.

**Reply by gockar** [reply]
Got the following response in https://codeberg.org/NextPush/nextpush-android/issues/4#issuecomment-1640029

> You both have buffering enabled.
> 
> For All-In-One, you have to add the following lines in nextcloud.conf:
> 
> ```
> <Proxy "fcgi://${NEXTCLOUD_HOST}:9000" enablereuse=on flushpackets=on max=10>
> </Proxy>
> ```
> It looks like this:
> 
> ```
> [...]
>     <FilesMatch "\.php$">
>         SetHandler "proxy:fcgi://${NEXTCLOUD_HOST}:9000"
>     </FilesMatch>
> 
>     <Proxy "fcgi://${NEXTCLOUD_HOST}:9000" enablereuse=on flushpackets=on max=10>
>     </Proxy> 
> [...]
> ```

Does that help? Really appreciate your help!

**Reply by szaimen** [reply]
I posted a comment: https://codeberg.org/NextPush/nextpush-android/issues/4#issuecomment-1640545
